### PR TITLE
Add Apache 2.0 LICENSE

### PR DESCRIPTION
**Pull request by punchagan**
The project has been "public" for a while now, but missing a LICENSE file. 

This commit adds a LICENSE file with the copyright year set to 2024. (I'm not sure if we want to set it to 2019, when the project first started).

### CODE REVIEW

**Comment by shonfeder** [comment]
It looks like we are missing licenses in many of the repos under this org. I've asked internally for some guidance on how we should proceed here. Excellent that you've surfaced this!

**Comment by punchagan** [comment]
I chose the Apache-2.0 license here based on the licenses used in other repos in this org, for instance [ocurrent], [obuilder], [solver-service], etc.

[ocurrent]: https://github.com/ocurrent/ocurrent
[obuilder]: https://github.com/ocurrent/obuilder
[solver-service]: https://github.com/ocurrent/solver-service

**Comment by shonfeder** [comment]
Hello, all contributors to this project! We have realized that no license was specified for this repo previously. We assume that the implicit intent of all contributors has always been to contribute their work under an open source license, but we need to make this official by receiving your consent to release your contributions. Could you please reply on this issue, or approve this PR, indicating that you approve the release of your contribution under the proposed Apache 2.0 license?

If you have any concerns or questions, please let us know. Thank you in advance for your help with this correction!

- [ ] @kit-ty-kate
- [ ] @talex5
- [ ] @benmandrew
- [ ] @craigfe
- [ ] @mtelvers
- [ ] @tmcgilchrist
- [ ] @shonfeder
- [ ] @avsm
- [ ] @MisterDA
- [ ] @Julow
- [ ] @punchagan
- [ ] @hannesm
- [ ] @art-w
- [ ] @dra27
- [ ] @moyodiallo
- [ ] @novemberkilo
- [ ] @emillon
- [ ] @gpetiot
- [ ] @mseri
- [ ] @patricoferris
- [ ] @vbmithr

**Review by benmandrew — APPROVED** [review]

**Review by gpetiot — APPROVED** [review]

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+      License. However, in accepting such obligations, You may act only
+      on Your own behalf and on Your sole responsibility, not on behalf
+      of any other Contributor, and only if You agree to indemnify,
+      defend, and hold each Contributor harmless for any liability
+      incurred by, or claims asserted against, such Contributor by reason
+      of your accepting any such warranty or additional liability.
+
+   END OF TERMS AND CONDITIONS
+
+   Copyright 2024 Thomas Leonard and others
```
Comment (kit-ty-kate) [inline LICENSE:178]: Having done the original fork from ocaml-ci and having written a significant portion of the codebase from then on until last year, i think i should probably appear somewhere on the copyright by name.
I also think the copyright should be backdated
```suggestion
   Copyright 2019-2024 Thomas Leonard, Kate Deplaix and others
```

**Review by talex5 — APPROVED** [review]

File: LICENSE (hunk 1 above)
Comment (punchagan) [reply LICENSE]: @kit-ty-kate I've updated the copyright line to be backdated to `2019` and included your name. 

(I used only the backdated year, and not an year range, as recommended on the Apache license page: https://www.apache.org/licenses/LICENSE-2.0)

File: LICENSE [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+      License. However, in accepting such obligations, You may act only
+      on Your own behalf and on Your sole responsibility, not on behalf
+      of any other Contributor, and only if You agree to indemnify,
+      defend, and hold each Contributor harmless for any liability
+      incurred by, or claims asserted against, such Contributor by reason
+      of your accepting any such warranty or additional liability.
+
+   END OF TERMS AND CONDITIONS
+
+   Copyright 2019 Thomas Leonard, Kate Deplaix and others
```
Comment (kit-ty-kate) [inline LICENSE:178]: A range of years is more adapted to maintained software
```suggestion
   Copyright 2019-2024 Thomas Leonard, Kate Deplaix and others
```

File: LICENSE (hunk 1 above)
Comment (punchagan) [reply LICENSE]: I was referring to the `[yyyy]`, and there isn't an explicit recommendation as you say. Sorry for the unclear previous message.

File: LICENSE (hunk 2 above)
Comment (punchagan) [reply LICENSE:178]: Wouldn't adding an year range add unnecessary busy work at the turn of every year; developers need to remember to update it each year and it's easy to miss, despite the software getting bug fixes, etc.? Isn't looking at commit activity on the repository a better indicator of how maintained the project is?

File: LICENSE (hunk 1 above)
Comment (punchagan) [reply LICENSE]: > For example see: https://github.com/apache/httpd/blob/trunk/LICENSE

I noticed that the project hasn't really filled up the license template and has just used the template directly (which is funny, given it's an apache project). (There are indeed other licenses with year ranges, though).

File: LICENSE (hunk 2 above)
Comment (kit-ty-kate) [reply LICENSE:178]: Some projects do update it, some don't. I'm not sure what's best and i don't think it matters much anyway, so feel free to ignore my suggestion. I think it's all a matter of personal preference

**Review by kit-ty-kate — APPROVED** [review]

**Comment by hannesm** [comment]
Sorry, I'm not a lawyer, and I don't quite understand what my (authors) things are for a (rather wordy) apache2 license? What are the benefits (for me as an author) for this license, compared to a permissive one (MIT/ISC)?

File: LICENSE (hunk 2 above)
Comment (punchagan) [reply LICENSE:178]: Thanks Kate! I think we'll go with just the year of applying the copyright to avoid adding additional work on the devs.

**Comment by shonfeder** [comment]
> What are the benefits (for me as an author) for this license, compared to a permissive one (MIT/ISC)?

@hannesm Please see https://github.com/ocurrent/opam-repo-ci/pull/346#issuecomment-2296561247 for the reason for this license selection.

Would you have contributed to this project if it had an Apache license at the time you made your contributions? Is so, would you be Ok rubber stamping this?

**Review by dra27 — APPROVED** [review]

**Review by mtelvers — APPROVED** [review]

**Review by moyodiallo — APPROVED** [review]

**Review by MisterDA — APPROVED** [review]

**Comment by shonfeder** [comment]
Note: please don't check your name off the checklist unless you also make your approval explicit via a PR review or comment. Unfortunately just checking the box will not leave a record of the approval. Thank you!

**Review by mseri — APPROVED** [review]

**Review by Julow — APPROVED** [review]

**Review by art-w — APPROVED** [review]

**Review by craigfe — APPROVED** [review]

**Review by novemberkilo — APPROVED** [review]

**Review by patricoferris — APPROVED** [review]

**Comment by shonfeder** [comment]
@hannesm, is your concern about the licenses blocking, or would you be willing to approve this PR to help unblock us?

**Comment by hannesm** [comment]
Sorry, I had this PR muted. @shonfeder indeed I usually think about licenses before contributing to a codebase. Since my time is limited, I won't dig into this "Apache 2.0 LICENSE", and just approve this PR.

**Review by hannesm — APPROVED** [review]

File: opam-repo-ci-api.opam [hunk 3]
```diff
@@ -7,6 +7,7 @@ authors: [
   "Craig Ferguson <me@craigfe.io>"
   "Anil Madhavapeddy <anil@recoil.org>"
 ]
+license: "Apache-2.0"
```
Comment (shonfeder) [inline opam-repo-ci-api.opam:10]: Just realizing we should ensure this makes it into all the opam files I think.

**Review by shonfeder — APPROVED** [review]

**Comment by shonfeder** [comment]
Thanks all!

**Pull request merged**